python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1209,3 +1209,113 @@ def empty_subset(
         cls, partitions_def: Optional[PartitionsDefinition] = None
     ) -> "DefaultPartitionsSubset":
         return cls()
+
+
+class AllPartitionsSubset(
+    NamedTuple(
+        "_AllPartitionsSubset",
+        [
+            ("partitions_def", PartitionsDefinition),
+            ("dynamic_partitions_store", Optional[DynamicPartitionsStore]),
```

Comment:
What if we made this class accept a `CachingInstanceQueryer` instead as a required arg?

This implementation means that new partitions can spring into existence and be included in the subset. If this class was used in the wrong context, some partitions may be included unintentionally. We could guard against this by accepting a `CachingInstanceQueryer` to enforce that this subset is used within the scope of a single "request" and contains an evaluation time.

Reply:
`CachingInstanceQueryer` is a dynamic partitions store and contains an evaluation time that could be used as current time

Reply:
yeah good point -- didn't think about the ability for things to change over time as a potential source of bugs (just wanted to keep things general)